docs/docs/how-to-build-custom-client/how-to-build-custom-client.md

```diff
@@ -0,0 +1,481 @@
+---
+title: How to build custom client
+---
+
+We usually use [next.js](https://nextjs.org/) to build our site client. This guide will show you how to build a custom client for your Comet DXP and render a typical content page from the page tree.
+
+### Page Tree Node List
+
+To load/see all page tree node's, you need to query the `pageTreeNodeList` field in the `Query` type. This field allows you to fetch a list of `PageTreeNode` objects based on specific criteria such as `contentScope` and `category`. Those values differ from project to project and are defined by the project's requirements.
+
+A Sample Page tree list can look like this:
+
+![Sample Page Tree List](images/sample-page-tree-list.png)
+
+The following example demonstrates how to query all page tree nodes and access their IDs, names, paths, and other relevant fields using fragments.
+
+To fetch all `PageTreeNode` objects and access their IDs, we will use the following query:
+
+```graphql
+query Pages($contentScope: PageTreeNodeScopeInput!, $category: String!) {
+    pageTreeNodeList(scope: $contentScope, category: $category) {
+        id
+        name
+        path
+        slug
+    }
+}
+```
+
+A Response will then look similar to this:
+
+```graphql
+{
+  "data": {
+    "pageTreeNodeList": [
+      {
+        "id": "7c151a7f-7e0c-4103-8d34-216421f4cdcf",
+        "name": "test",
+        "path": "/test",
+        "slug": "test"
+      }
+    ]
+  }
+}
+```
+
+With that information in place we can now query the page content, based on the pageTreeNode id.
+
+## Working with the Comet's GraphQL API
+
+The following example demonstrates a Content Page that includes a `HeadlineBlock`, `ImageBlock`, `RichTextBlock`, and a `ColumnBlock` containing two `ImageBlock` elements.
+
+![Sample Content Page](images/sample-content-page.png)
+
+### Loading page data from the API
```

Comment:
I'd strongly suggest mentioning the page tree node by path approach here.

Reply:
done.